core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -282,3 +282,80 @@ getpeername(int sockfd, struct sockaddr *addr, socklen_t *addrlen)
 
     return __WASI_ERRNO_SUCCESS;
 }
+
+struct timeval
+time_us_to_timeval(uint64_t time_us)
+{
+    struct timeval tv;
+    tv.tv_sec = time_us / 1000000UL;
+    tv.tv_usec = time_us % 1000000UL;
+    return tv;
+}
+
+uint64_t
+timeval_to_time_us(struct timeval tv)
+{
+    return (tv.tv_sec * 1000000UL) + tv.tv_usec;
+}
+
+int
+getsockopt(int sockfd, int level, int optname, void *__restrict optval,
+           socklen_t *__restrict optlen)
+{
+    __wasi_errno_t error;
+    uint64_t timeout_us;
+
+    switch (level) {
+        case SOL_SOCKET:
+            switch (optname) {
+                case SO_RCVTIMEO:
+                    assert(*optlen == sizeof(struct timeval));
+                    error = __wasi_sock_get_recv_timeout(sockfd, &timeout_us);
+                    HANDLE_ERROR(error);
+                    *(struct timeval *)optval = time_us_to_timeval(timeout_us);
+                    return error;
+                    break;
```

Comment:
there's no need for `break` if you already have `return` one line above that.

Reply:
Forgot to remove, thanks for catching